lib/router/router.js

```diff
@@ -260,7 +260,12 @@ export default class Router extends EventEmitter {
     const jsonPageRes = await this.fetchRoute(route)
     const jsonData = await jsonPageRes.json()
 
-    if (jsonData.buildIdMismatch) {
+    const { buildId } = window.__NEXT_DATA__
+
+    if (
+      process.env.NODE_ENV === 'production' &&
+      buildId !== jsonPageRes.headers.get('Next-Build-Id')
```

Comment:
@rauchg By looking at all these, it's seem like keeping `/_next/pages/BUILD_ID/<path>` is a better solution than above options.

Both ^^^ options (or simply removing BUILD_ID) from the URL won't give us much benefit. Instead, we won't be able to do immutable caching.